src/bindings/node/node.cc

```diff
@@ -0,0 +1,196 @@
+#include <assert.h>
+#include <boost/make_shared.hpp>
+#include <boost/noncopyable.hpp>
+#include <boost/optional.hpp>
+#include <boost/property_tree/json_parser.hpp>
+#include <boost/property_tree/ptree.hpp>
+#include <boost/shared_ptr.hpp>
+#include <iostream>
+#include <node_api.h>
+#include <sstream>
+#include <string>
+
+#include "valhalla/tyr/actor.h"
+
+boost::property_tree::ptree json_to_pt(const char* json) {
+  std::stringstream ss;
+  ss << json;
+  boost::property_tree::ptree pt;
+  boost::property_tree::read_json(ss, pt);
+  return pt;
+}
+
+boost::property_tree::ptree make_conf(const char* config) {
+  return json_to_pt(config);
+}
+
+// check if napi_status is nap_ok, throw error if not
+void checkNapiStatus(napi_status status, napi_env env, const char* error_message) {
+  if (status != napi_ok) {
+    napi_throw_error(env, NULL, error_message);
+  }
+}
+
+#define DECLARE_NAPI_METHOD(name, func)                                                              \
+  { name, 0, func, 0, 0, 0, napi_default, 0 }
+
+class Actor {
+public:
+  static napi_value Init(napi_env env, napi_value exports) {
+    napi_status status;
+    napi_property_descriptor properties[] = {
+        DECLARE_NAPI_METHOD("route", Route),
+    };
+
+    napi_value cons;
+    status = napi_define_class(env, "Actor", NAPI_AUTO_LENGTH, New, nullptr, 1, properties, &cons);
+    assert(status == napi_ok);
+
+    status = napi_create_reference(env, cons, 1, &constructor);
```

Comment:
reading https://nodejs.org/api/n-api.html#n_api_napi_create_reference it looks like `&constructor` is a reference to the result. Let's rename it to `router`? :)

Reply:
ultimately, it will be able to do more than route - it should be able to match, etc.